open-craft/problem-builder
# New version of mentoring block that supports explicit steps

This PR introduces two new blocks and enables support for viewing and editing them in Studio. One of the blocks is a new version of `MentoringBlock` that supports explicit steps, the other one is a block that represents a step:

-  The new mentoring block only allows step blocks (any number) and mentoring-wide messages as direct children.
-   The step block hosts all blocks that can not be added to new mentoring block (MCQ, MRQ, Long Answer, etc.)

**Acceptance criteria**

Studio integration:

-   [X] Make it possible to add new mentoring block to unit
-   [X] Make it possible to **add** step blocks
-   [X] Make it possible to **edit** step blocks
-   [X] Make it possible to **remove** step blocks
-   [X] Make it possible to **add step contents**
-   [X] Previews at mentoring and step levels should function properly
    -   [X] inner blocks should not have a studio header (small glitch: at mentoring level, HTML block still has a header)
    -   [X] inner blocks should look more or less the same as they would in LMS
    -   [X] inner blocks should have no obvious styling glitches and pass other common-sense checks

**Testing**

1. Enable new mentoring block for a course in Studio by adding `pb-mentoring` to advanced modules.
2. Add new mentoring block to a (new or existing) unit by choosing "Mentoring Questions (with explicit steps)" from the list of "Advanced" blocks.
3. Click "View" to edit the components of the block. Observe that the options for blocks to add are restricted to step and message blocks.
4. Add a step block.
5. Click "View" to edit the components of the block. Observe that the options for blocks to add include all relevant blocks that can not be added to the top level of the new mentoring block.
6. Try adding and removing a few components to/from the step block.
7. On the previous (mentoring) level, click "Edit" to change e.g. the title of the step block.
8. Click the trash icon to remove the step block.
